### PR TITLE
feat: implement checkbox

### PR DESCRIPTION
**Pull request by amirhossein-asadi**

### CODE REVIEW

File: libs/core/src/styles/createTheme.tsx [hunk 1]
```diff
@@ -2,7 +2,10 @@ import { loadingButtonClasses } from '@mui/lab';
 import '@mui/lab/themeAugmentation';
 import { alpha, createTheme as createMuiTheme, Theme, ThemeOptions } from '@mui/material';
 import { buttonBoxShadowAlpha, buttonDisableAlpha } from '../button/constants';
-import { background, common, defaultPalette } from '../colors';
+import { Checked } from '../checkbox/Icons/Checked';
+import { Indeterminate } from '../checkbox/Icons/Indeterminate';
+import { Unselected } from '../checkbox/Icons/Unselected';
```
Comment (mdeljavan) [inline libs/core/src/styles/createTheme.tsx:7]: remove

File: libs/core/src/styles/createTheme.tsx [hunk 2]
```diff
@@ -366,6 +381,53 @@ const defaultOptions: ThemeOptions = {
...
+          padding: 0,
+          '& .MuiSvgIcon-root.MuiSvgIcon-fontSizeSmall': {
+            fontSize: '1rem', // 16px
+          },
+          '&& .MuiSvgIcon-root.MuiSvgIcon-fontSizeMedium': {
+            fontSize: '1.25rem', // 20px
+          },
+          '& .MuiSvgIcon-root.MuiSvgIcon-fontSizeLarge': {
+            fontSize: '1.5rem', // 24px
+          },
```
Comment (mdeljavan) [inline libs/core/src/styles/createTheme.tsx:403]: remove these and override fontSize in MuiSvgIcon

**Review by mdeljavan — CHANGES_REQUESTED** [review]

File: libs/core/src/checkbox/Icons/Checked.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { SvgIcon, SvgIconProps } from '@mui/material';
+
+export function Checked(props: SvgIconProps) {
+  return (
+    <SvgIcon {...props}>
```
Comment (mdeljavan) [inline libs/core/src/checkbox/Icons/Checked.tsx:5]: set viewBox

File: libs/core/src/checkbox/checkbox.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        '&.MuiSvgIcon-fontSizeSmall': {
+          fontSize: '1rem',
+        },
+        '&.MuiSvgIcon-fontSizeMedium': {
+          fontSize: '1.25rem',
+        },
+        '&.MuiSvgIcon-fontSizeLarge': {
+          fontSize: '1.5rem',
+        },
+      },
```
Comment (mdeljavan) [inline libs/core/src/checkbox/checkbox.ts:34]: define svg font size in MuiSvgIcon component

**Review by mdeljavan — CHANGES_REQUESTED** [review]

**Review by mdeljavan — DISMISSED** [review]

**Review by alisajadih — APPROVED** [review]

**Review by mdeljavan — APPROVED** [review]

**Review by mdeljavan — APPROVED** [review]

**Pull request merged**